packages/react-charting/src/components/PieChart/Arc/Arc.tsx

```diff
@@ -1,49 +1,131 @@
   };
 
   public render(): JSX.Element {
-    const { color, arc } = this.props;
+    const { arc } = this.props;
     const getClassNames = classNamesFunction<IArcProps, IArcStyles>();
-    const classNames = getClassNames(getStyles, { color });
-    return <path d={arc(this.props.data)} className={classNames.root} onClick={this.props.data?.data.onClick} />;
+    const classNames = getClassNames(props => getStyles(props, this.props.theme), { ...this.props });
+
+    return (
+      <path
+        d={arc(this.props.data)}
+        className={`${this.state.isArcFocused ? classNames.pieRootFocused : classNames.pieRoot}`}
+        onClick={this.props.data?.data.onClick}
+      />
+    );
   }
+
+  protected _onFocus = () => {
+    this.setState({ isArcFocused: true });
+  };
+
+  protected _onBlur = () => {
+    this.setState({ isArcFocused: false });
+  };
 }
 
 export class LabeledArc extends Arc {
+  public constructor(props: IArcProps) {
+    super(props);
+    this.state = {
+      id: getId(),
+    };
+  }
+
   public render(): JSX.Element {
-    const { data, arc, culture } = this.props;
-    const [labelX, labelY] = arc.centroid(data);
-    const labelTranslate = `translate(${labelX}, ${labelY})`;
+    const { data, culture } = this.props;
+    const gap = 4;
+    // placing the labels on the outside arc
+    const [labelX, labelY] = shape.arc().centroid({
+      endAngle: data?.endAngle || 0,
+      startAngle: data?.startAngle || 0,
+      padAngle: data?.padAngle,
+      innerRadius: this.props.outerRadius,
+      outerRadius: this.props.outerRadius + gap,
+    });
+
+    const getClassNames = classNamesFunction<IArcProps, IArcStyles>();
+    const classNames = getClassNames(props => getStyles(props, this.props.theme));
+
+    const angle = ((data?.startAngle || 0) + (data?.endAngle || 0)) / 2;
+
+    const content = `${data?.data.x}-${convertToLocaleString(data?.data.y, culture)}`;
 
     return (
-      <g className="arc">
+      <g
+        className={classNames.pie}
+        data-is-focusable={true}
+        id={this.state.id}
+        onFocus={this._onFocus}
+        onBlur={this._onBlur}
+        aria-label={content}
+        role="img"
+      >
         {super.render()}
-        <text transform={labelTranslate} textAnchor="middle">
-          {data!.data!.x}-{convertToLocaleString(data!.data!.y, culture)}
-        </text>
+        <SVGTooltipText
+          content={content}
+          textProps={{
+            x: labelX,
+            y: labelY,
+            dominantBaseline: angle > Math.PI / 2 && angle < (3 * Math.PI) / 2 ? 'hanging' : 'auto',
+            textAnchor: angle > Math.PI ? 'end' : 'start',
+            'aria-label': `${data?.data.x}-${convertToLocaleString(data?.data.y, culture)}`,
+            className: classNames.pieText,
+          }}
+          showTooltip={this.state.isArcFocused}
+          shouldReceiveFocus={false}
+          maxWidth={40}
+          wrapContent={this._wrapContent}
```

Comment:
This function is returning overflow, not wrapping behavior

Reply:
This is a utility function being used in donut chart as well. The wrapContent will only truncate the svg text, when it's overflowing (beyond the max width), so in the function we're computing if it's overflowing or now


with wrapContent.
![image](https://github.com/microsoft/fluentui/assets/70366079/37236479-ecf7-4e36-80df-d8e9b31893e5)

w/o wrapContent
![image](https://github.com/microsoft/fluentui/assets/70366079/fc111df3-9c01-4172-a093-e68ce24ef130)

Reply:
This is a utility function being used in donut chart as well. The wrapContent will only truncate the svg text, when it's overflowing (beyond the max width), so in the function we're computing if it's overflowing or now


with wrapContent.
![image](https://github.com/microsoft/fluentui/assets/70366079/37236479-ecf7-4e36-80df-d8e9b31893e5)

w/o wrapContent
![image](https://github.com/microsoft/fluentui/assets/70366079/fc111df3-9c01-4172-a093-e68ce24ef130)